community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -54,20 +53,101 @@
+        run(f"{lookup().scontrol} update nodename={hostlist} state=power_up")
+
+class NodeActionIdle(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        """idle nodes"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.info(f"{count} nodes to idle ({hostlist})")
+        run(f"{lookup().scontrol} update nodename={hostlist} state=resume")
+
+class NodeActionRestart(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        """start instances for nodes"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.info(f"{count} instances restarted ({hostlist})")
+        start_instances(nodes)
+
+class NodeActionDown(NodeAction):
+    def __init__(self, reason:str="Instance stopped/deleted"):
+        self.reason = reason
+
+    def __hash__(self):
+        return hash((type(self), self.reason))
+
+    def __eq__(self, other):
+        return (type(self) == type(other)) and (self.reason == other.reason)
+
+    def apply(self, nodes: List[str]) -> None:
+        """down nodes"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.info(
+            f"{count} nodes set down ({hostlist})"
+        )
+        run(
+            f"{lookup().scontrol} update nodename={hostlist} state=down reason='{self.reason}'"
+        )
+
+class NodeActionPowerDown(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        """power_down node in slurm"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.info(f"{count} instances to power down ({hostlist})")
+        run(f"{lookup().scontrol} update nodename={hostlist} state=power_down")
+
+class NodeActionDelete(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        """delete instances for nodes"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.info(f"{count} instances to delete ({hostlist})")
+        delete_instances(nodes)
+
+class NodeActionUnknown(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        """Error status, nodes shouldn't get in this status"""
+        count = len(nodes)
+        hostlist = util.to_hostlist_fast(nodes)
+        log.error(f"{count} nodes have unexpected status: ({hostlist})")
+        first = next(iter(nodes))
+        state = lookup().slurm_node(first)
+        state = "{}+{}".format(state.base, "+".join(state.flags)) if state else "None"
+        inst = lookup().instance(first)
+        log.error(f"{first} state: {state}, instance status:{inst.status}")
+
+class NodeActionPrempt(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        NodeActionDown("Preempted instance").apply(nodes)
+        NodeActionRestart().apply(nodes)
+
+class NodeActionUnchanged(NodeAction):
+    def apply(self, nodes:List[str]) -> None:
+        pass
+
+class NodeActionUnback(NodeAction):
```

Comment:
```py
NODE_ACTION_UNBACK = NodeActionDown("Unbacked instance")
```

Reply:
Now that `reason` doesn't have a default value couldn't we just define it inline (sub old `NodeActionUnback` instances directly NodeActionDown("Unbacked instance"))?